pkg/queue/readiness/probe.go

```diff
@@ -168,7 +168,7 @@ func (p *Probe) doProbe(probe func(time.Duration) error) error {
 		return p.count >= p.SuccessThreshold, nil
 	})
 
-	if lastProbeErr != nil {
+	if pollErr != nil && lastProbeErr != nil {
```

Comment:
It might be a better idea to reset the `lastProbeError` on every retry and ignore the `pollErr` altogether.

I.e.

```golang
wait.PollImmediate(retryInterval, p.pollTimeout, func() (bool, error) {
  lastPollErr = probe(aggressiveProbeTimeout)
  if lastPollErr != nil {
    ...
```

I think that's less confusing and better transports the true meaning of the error, also to clients of this function, seeing as we're currently returning the `timed out` error, which usually tells you not a ton.

Reply:
the problem with that is successThreshold, it's possible to succeed, but not enough times in a row in which case lastProbeError could be nil :/

Reply:
ahhh, good point. I guess the timeout error is the correct one in that case.